### ISSUE TITLE
Endgame Combat turret range upgrade prevents blueprinting from working

### ISSUE THREAD

**Issue by Baughn** [ReikaKalseki/Reika_FactorioMods_Issues#44]
Specifically, the range boost is embedded inside the blueprint somehow. The turrets appear in the blueprint, but cannot be placed using one.

**Comment by ReikaKalseki** [comment]
I fixed that months ago, and confirmed it to work.

**Comment by Baughn** [comment]
This was with the most recent version installed through the ingame mod browser. I wouldn't even know where to get an old one.

Are you sure it still works?

**Comment by ReikaKalseki** [comment]
I have not changed anything, and Factorio has not updated. Try removing other mods.

**Comment by Baughn** [comment]
I was able to reproduce this with only Endgame Combat installed.

This is on Factorio 0.15.34 (build 30875, win64, alpha).

**Comment by Baughn** [comment]
Thinking about this a bit more, it still could be caused by a different mod. I tested it by removing everything else from an existing save; I didn't test it with a fresh game, as that'd have taken much longer.

I can do so if you want me to, though. Not sure how plausible it is for that to be the cause, as I did build and emplace fresh turrets for the test.

**Comment by ReikaKalseki** [comment]
Please try.

**Comment by Baughn** [comment]
Same problem.

However, while testing this in creative mode, I noticed that the blueprint only fails to place for the first couple of techs. After unlocking *every* tech, although the blueprint component list shows the same 'unknown key' error, the blueprinted turrets nevertheless place correctly.

**Comment by ReikaKalseki** [comment]
What about the range boost techs specifically?

**Comment by Baughn** [comment]
Sorry, since then I've pulled that computer apart for a rebuild. If you want me to check specifically which techs are making it work, you'll have to wait a few days.

**Comment by ReikaKalseki** [comment]
That is fine.

**Comment by Baughn** [comment]
It's the "construction robotics" tech that makes it work. Interesting, that; IIRC, it used to be necessary to research it for blueprint ghosts to work at all.

**Comment by ReikaKalseki** [comment]
So in other words, the issue is actually of no ingame impact?

**Comment by Baughn** [comment]
No, I use blueprint ghosts to plan out my base right from the start. There are also a multitude of mods that allow for automated construction without that particular technology.
